main.tf

```diff
@@ -20,11 +20,9 @@ terraform {
 
   }
 }
-provider "aws" {
```

Comment:
The other providers helm and kubernetes have to be removed as well. I suggest so swap them out to a file named providers.tf.template with the following contents:

```hcl
provider "aws" {
  profile             = "your profile" # or maybe var.profile?
}
provider "kubernetes" {
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  token                  = data.aws_eks_cluster_auth.cluster.token
}

provider "helm" {
  kubernetes {
    host                   = data.aws_eks_cluster.cluster.endpoint
    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
    token                  = data.aws_eks_cluster_auth.cluster.token
  }
}

# + All necessary data objects
```

Readme.md must be updated accordingly. Users who use the reference architecture as a root module just can rename providers.tf.example to providers.tf and fill in the profile name. User who use the reference architecture as a submodule can copy (and merge) it to their root-level definition. I have seen this pattern in other third party modules.

Reply:
What does this have to do with the initial change for getting rid of specifying redundant AWS account id?

In my opinion, this should be done in a separate change, because i assume, that there will be more related changes required, e.g. on how to specify certificate information best for the different usage options you want to introduce.

Reply:
I am fine if we remove all providers in a separate PR, but we should remove them all at once. So please then revert the AWS provider for this PR.

Reply:
Ok, i reverted the changes, please review again